Generals/Code/GameEngine/Source/GameNetwork/GameInfo.cpp

```diff
@@ -907,25 +922,42 @@ AsciiString GameInfoToAsciiString( const GameInfo *game )
 	}
 
 	AsciiString optionsString;
+#if RTS_GENERALS
 	optionsString.format("M=%2.2x%s;MC=%X;MS=%d;SD=%d;C=%d;", game->getMapContentsMask(), newMapName.str(),
 		game->getMapCRC(), game->getMapSize(), game->getSeed(), game->getCRCInterval());
+#else
+	optionsString.format("US=%d;M=%2.2x%s;MC=%X;MS=%d;SD=%d;C=%d;SR=%u;SC=%u;O=%c;", game->getUseStats(), game->getMapContentsMask(), newMapName.str(),
+		game->getMapCRC(), game->getMapSize(), game->getSeed(), game->getCRCInterval(), game->getSuperweaponRestriction(),
+		game->getStartingCash().countMoney(), game->oldFactionsOnly() ? 'Y' : 'N' );
+#endif
+
+	//add player info for each slot
 	optionsString.concat(slotListID);
 	optionsString.concat('=');
 	for (Int i=0; i<MAX_SLOTS; ++i)
 	{
 		const GameSlot *slot = game->getConstSlot(i);
+
 		AsciiString str;
 		if (slot && slot->isHuman())
 		{
-			AsciiString name = WideCharStringToMultiByte(slot->getName().str()).c_str();
-
-			str.format( "H%s,%X,%d,%c%c,%d,%d,%d,%d,%d:",
-				name.str(), slot->getIP(), slot->getPort(),
+			AsciiString tmp;  //all this data goes after name
+			tmp.format( ",%X,%d,%c%c,%d,%d,%d,%d,%d:",
+				slot->getIP(), slot->getPort(),
 				(slot->isAccepted()?'T':'F'),
 				(slot->hasMap()?'T':'F'),
 				slot->getColor(), slot->getPlayerTemplate(),
 				slot->getStartPos(), slot->getTeamNumber(),
 				slot->getNATBehavior() );
+			//make sure name doesn't cause overflow of m_lanMaxOptionsLength
+			int lenCur = tmp.getLength() + optionsString.getLength() + 2;  //+2 for H and trailing ;
+			int lenRem = m_lanMaxOptionsLength - lenCur;  //length remaining before overflowing
```

Comment:
Can this be negative? Maybe use` int lenMax = max(0, lenRem / (MAX_SLOTS - i));`